src/Vendoring/OpenTelemetry.Instrumentation.SqlClient/SqlClientTraceInstrumentationOptions.cs

```diff
@@ -0,0 +1,303 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+#nullable disable
+
+using System.Collections.Concurrent;
+using System.Data;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Instrumentation.SqlClient;
+
+/// <summary>
+/// Options for <see cref="SqlClientInstrumentation"/>.
+/// </summary>
+/// <remarks>
+/// For help and examples see: <a href="https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.SqlClient/README.md#advanced-configuration" />.
+/// </remarks>
+public class SqlClientTraceInstrumentationOptions
```

Comment:
These need to be `internal`. We don't want to ship these as public from our library.

Reply:
One suggestion for future releases, consider leveraging the public API analyzer in CI.

* https://github.com/open-telemetry/opentelemetry-dotnet/pull/806
* https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Api/.publicApi/Stable/PublicAPI.Shipped.txt